packages/eslint-plugin/test/__file_snapshots__/types/expect/index.d.ts.lint

```diff
@@ -0,0 +1,19 @@
+types/expect/index.d.ts
+  7:12  error  Array type using 'readonly T[]' is forbidden for non-simple types. Use 'ReadonlyArray<T>' instead  @typescript-eslint/array-type
+
+✖ 1 problem (1 error, 0 warnings)
+  1 error and 0 warnings potentially fixable with the `--fix` option.
+
+==== types/expect/index.d.ts ====
+
+    export const foo = 1234;
+    
+    export const aUnion: string | number | undefined;
+    
+    export function complicatedUnion<T extends string | number>(x: T, y: T): {
+        prop1: "a" | "b" | "c";
+        prop2: readonly (string | number)[];
```

Comment:
```suggestion
        prop2: ReadonlyArray<string | number>;
```

Reply:
This is intentional to test that I can use one syntax or the other in the rule that normalizes the form.